src/js/renderables/allies/adult.js

```diff
@@ -0,0 +1,64 @@
+import * as me from 'melonjs';
+import game from '../../game.js';
+
+class AdultEntity extends me.Entity {
+
+    constructor(x, y, settings) {
+        // call the parent constructor
+        super(x, y , settings);
+
+        // set default stats of adult unit
+        this.updateAllyStats()
+    }
+
+    getAllyStats() {
+        return {
+            allyTier: this.tier,
+            allyCost: this.allyCost,
+            allyATK: this.allyATK,
+            allyASPD: this.allyASPD,
+            allyRange: this.allyRange
+        }
+    }
+
+    updateAllyStats() {
+        if (this.tier < 4) {
```

Comment:
Can you explain to me why we have a nested if statement?
It is kinda confusing to me.

Reply:
Technically the tier should never go past 3, but I did use an else statement on line 38, so I thought it would be better to check it. I'll simplify it to just an else-if.

Reply:
Also realized I forgot to update the parent class of a few files--fixing that now.